src/viam/ur/module/ur_arm_state_controlled.cpp

```diff
@@ -62,56 +62,97 @@ std::optional<URArm::state_::event_variant_> URArm::state_::state_controlled_::h
+        if (mr.cancellation_request) {
+            std::exchange(state.move_request_, {})->complete_cancelled();
+            return std::nullopt;
+        }
 
-    } else if (state.move_request_->samples.empty() && state.move_request_->cancellation_request &&
-               !state.move_request_->cancellation_request->issued) {
-        // We have a move request, the samples have been forwarded,
-        // and cancellation is requested but has not yet been issued. Issue a cancel.
-        state.move_request_->cancellation_request->issued = true;
         if (!arm_conn_->driver->writeTrajectoryControlMessage(
-                urcl::control::TrajectoryControlMessage::TRAJECTORY_CANCEL, 0, RobotReceiveTimeout::off())) {
-            state.move_request_->cancel_error("failed to write trajectory control cancel message to URArm");
-            VIAM_SDK_LOG(error) << "While in state " << describe()
-                                << ", failed to write trajectory control cancel message; dropping connection";
+                urcl::control::TrajectoryControlMessage::TRAJECTORY_START, 1, RobotReceiveTimeout::off())) {
+            VIAM_SDK_LOG(error) << "single pose: start failed; dropping connection";
+            std::exchange(state.move_request_, {})->complete_error("failed to send trajectory start for single pose");
             return event_connection_lost_::trajectory_control_failure();
         }
-    } else if (!state.move_request_->samples.empty() && state.move_request_->cancellation_request) {
-        // We have a move request that we haven't issued but a
-        // cancel is already pending. Don't issue it, just cancel it.
-        std::exchange(state.move_request_, {})->complete_cancelled();
-    } else {
-        // TODO: is it assured that we have positions/velocities here?
-        state.move_request_->write_joint_data(state.ephemeral_->joint_positions, state.ephemeral_->joint_velocities);
+
+        auto ps = std::move(*pose_cmd);  // dereference the pointer, now it is empty and we have the data
+
+        VIAM_SDK_LOG(info) << "URArm sending single pose (tool-space)";
+
+        // TODO: velocity and acceleration should not be hardcoded
+        // gotcha is to not trip the path sanity error
+        const float velocity = 0.25F;
+        const float acceleration = 0.5F;
+        const float timestep = 0;
+        const float blend_radius = 0;
+        const bool cartesian = true;
+
+        if (!arm_conn_->driver->writeTrajectoryPoint(ps.p, acceleration, velocity, cartesian, timestep, blend_radius)) {
+            VIAM_SDK_LOG(error) << "single pose: writeTrajectoryPoint failed; dropping connection";
+            std::exchange(state.move_request_, {})->complete_error("failed to send single pose point");
+            return event_connection_lost_::trajectory_control_failure();
+        }
+
+        // Mark as sent by replacing with empty vector
+        mr.move_command = std::vector<trajectory_sample_point>{};
```

Comment:
Tricky! It does work but maybe it is a little too tricky?

If it is too tricky, you could have the variant hold a `std::optional<pose_sample>`. That'd let you swap it out for `std::nullopt` when you consume it.

Honestly, that's *probably* a better choice. I feel like having the pose handling code suddenly convert itself into an issued trajectory move is going to cause some puzzlement at debugging time someday. It'll look like it was a trajectory move, but it is actually a post-issue move_to_pose.